code/game/objects/items/weapons/RFD.dm

```diff
@@ -145,135 +148,114 @@ RFD Construction-Class
 		if("Floors and Walls")
-			mode = RFD_FLOORS_AND_WALL
-		if("Windows and Grille")
-			mode = RFD_WINDOWS_AND_GRILLE
+			mode = 1
 		if("Airlock")
-			mode = RFD_AIRLOCK
+			mode = 2
 		if("Deconstruct")
-			mode = RFD_DECONSTRUCT
+			mode = 3
 		else
-			mode = RFD_FLOORS_AND_WALL
+			mode = 1
 	if(current_mode)
 		to_chat(user, SPAN_NOTICE("You switch the selection dial to <i>\"[current_mode]\"</i>."))
-		playsound(src.loc, 'sound/effects/pop.ogg', 50, 0)
+		if(mode == 3)
+			playsound(get_turf(src), 'sound/weapons/laser_safetyoff.ogg', 50, FALSE)
+		else
+			playsound(get_turf(src), 'sound/weapons/laser_safetyon.ogg', 50, FALSE)
 		if(prob(20))
 			spark(get_turf(src), 3, alldirs)
 
 /obj/item/rfd/construction/afterattack(atom/A, mob/user, proximity)
 	if(!proximity)
 		return
-	if((!isturf(A) && !isturf(A.loc)) || istype(A, /obj/structure/table))
-		return
 	if(disabled && !isrobot(user))
 		return FALSE
 	if(istype(get_area(A),/area/shuttle)||istype(get_area(A),/turf/space/transit))
 		return FALSE
 	var/turf/t = get_turf(A)
-	if(isNotStationLevel(t.z))
+	if (isNotStationLevel(t.z))
 		return FALSE
-	if(is_type_in_list(A, valid_atoms))
-		return alter_atom(A, user, (mode == RFD_DECONSTRUCT))
-	return alter_atom(t, user, (mode == RFD_DECONSTRUCT))
+	return alter_turf(A, user, (mode == 3))
+
+/obj/item/rfd/construction/proc/alter_turf(var/turf/T,var/mob/user,var/deconstruct)
 
-/obj/item/rfd/construction/proc/alter_atom(var/atom/A, var/mob/user, var/deconstruct)
 	if(working)
 		return FALSE
 
-	var/turf/T = isturf(A) ? A : null // the lower istypes will return false if T is null, which means we don't have to check whether it's an atom or a turf
-	if(mode == RFD_FLOORS_AND_WALL)
-		if(istype(T, /turf/space) || istype(T, T.baseturf))
-			build_cost =  1
-			build_type =  "floor"
-			build_atom =  /turf/simulated/floor/airless
-		else if(istype(T, /turf/simulated/floor))
-			build_delay = 20
-			build_cost =  3
-			build_type =  "wall"
-			build_atom =  /turf/simulated/wall
-	else if(mode == RFD_WINDOWS_AND_GRILLE)
-		if(istype(T, /turf/simulated/floor))
-			build_cost =  3
-			build_delay = 20
-			build_type = "window and grille section"
-			build_atom = /obj/effect/wingrille_spawn/reinforced
-	else if(mode == RFD_AIRLOCK)
-		if(istype(T, /turf/simulated/floor))
-			build_cost =  3
-			build_delay = 20
-			build_type = "airlock"
-			build_atom = /obj/machinery/door/airlock
-	else if(mode == RFD_DECONSTRUCT)
-		if(istype(A, /obj/machinery/door/airlock))
-			build_cost =  10
-			build_delay = 50
-			build_type = "airlock"
-		else if(istype(T, /turf/simulated/wall))
-			var/turf/simulated/wall/W = T
-			build_delay = 50
-			build_cost =  5
-			build_type =  (!canRwall && W.reinf_material) ? null : "wall"
-			build_atom =  W.under_turf
-		else if(istype(T, /turf/simulated/floor))
-			build_delay = 50
-			build_cost =  10
-			build_type =  "floor"
-			build_atom =  T.baseturf
+	if(mode == 3 && istype(T,/obj/machinery/door/airlock))
```

Comment:
The rfd modes was moved from using defines back to numbers? I assume this is due to a faulty merge, going back to a previous version of the code.